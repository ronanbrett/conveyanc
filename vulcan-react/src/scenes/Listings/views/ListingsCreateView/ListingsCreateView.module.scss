@import "src/styles/vars";

.ListingsCreateView {

}