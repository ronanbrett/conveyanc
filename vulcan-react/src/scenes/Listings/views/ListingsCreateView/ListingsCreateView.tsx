import { PropertyInfo, PropertyType } from "@core/api/graphql";
import { getQueryRXJS } from "@core/utils/rxjs.utils";
import ListingCreateForm from "@scenes/Listings/components/ListingCreateForm";
import { RETRIEVE_PROPERTY_INFO } from "@scenes/Listings/Listings.queries";
import { TopNavPortal, Button } from "components";
import React, { useEffect, useState } from "react";
import { map, take } from "rxjs/operators";

import styles from "./ListingsCreateView.module.scss";

interface ListingsCreateViewProps {
  children?: any;
}

interface ListingCreateViewState extends PropertyInfo {
  isReady: boolean;
}

const ListingsCreateView = (props: ListingsCreateViewProps) => {
  const [fieldData, setFieldData] = useState<ListingCreateViewState>({
    propertyType: [],
    isReady: false,
  });

  function activateLasers() {
    console.log("The link was clicked.");
  }

  useEffect(() => {
    // const info = async () =>
    //   await getQueryRXJS(RETRIEVE_PROPERTY_INFO).pipe(take(1)).toPromise();

    // info();

    const fetchData = async () => {
      const result = await getQueryRXJS(RETRIEVE_PROPERTY_INFO)
        .pipe(
          take(1),
          map((x: { propertyInfo: PropertyInfo }) => x.propertyInfo)
        )
        .toPromise();

      setFieldData({ ...result, isReady: true });
    };

    fetchData();
  }, []);

  //

  return (
    <div className={`${styles.ListingsCreateView} page`}>
      <TopNavPortal>
        <Button onClick={activateLasers}>Save</Button>
      </TopNavPortal>
      {fieldData.isReady ? (
        <ListingCreateForm fieldData={fieldData}></ListingCreateForm>
      ) : (
        <></>
      )}
    </div>
  );
};

export default ListingsCreateView;
