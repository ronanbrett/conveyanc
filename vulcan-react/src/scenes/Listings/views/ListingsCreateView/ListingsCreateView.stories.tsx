import React from 'react';

import ListingsCreateView from './ListingsCreateView';

import { action } from '@storybook/addon-actions';

export default {
  title: 'Components/ListingsCreateView',
  decorators: [],
  component: ListingsCreateView,
};


export const Default = () => (
  <ListingsCreateView></ListingsCreateView>
);
