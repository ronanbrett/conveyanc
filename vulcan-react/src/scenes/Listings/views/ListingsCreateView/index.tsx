import ListingsCreateView from './ListingsCreateView';

export default ListingsCreateView;