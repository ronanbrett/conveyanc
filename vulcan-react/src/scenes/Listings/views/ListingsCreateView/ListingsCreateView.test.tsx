import { render } from "@testing-library/react";
import { MockRouter, createTopNavPortalMock } from "testing";
import React from "react";
import ListingsCreateView from "./ListingsCreateView";

createTopNavPortalMock();

const TestViewComponent = () => <MockRouter component={ListingsCreateView} />;

describe("ListingsCreateView", () => {
  it("renders without error", () => {
    const { asFragment } = render(<TestViewComponent />);
    expect(asFragment()).toMatchSnapshot();
  });
});
