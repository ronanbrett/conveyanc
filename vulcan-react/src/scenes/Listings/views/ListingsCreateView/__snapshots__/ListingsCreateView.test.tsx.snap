// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingsCreateView renders without error 1`] = `
<DocumentFragment>
  <div
    class="ListingsCreateView page"
  />
</DocumentFragment>
`;
